### PR TITLE
Use Akka Streams instead of Akka Actors

### PR DESCRIPTION
**Pull request by wsargent**
This PR moves the reactive stocks example to use streams instead of actors as the underlying mechanic.  The underlying code is the same as the Scala version, only with some tweaks necessitated by Jackson's serialization.

One thing that was surprising is that the websocket client ignores the value set for "Origin" and instead sends the same value as serverURL.  It fails because it's on a different port rather than because it's checking the header and finding a difference.

Fixes https://github.com/playframework/play-java-websocket-example/issues/46

Supersedes https://github.com/playframework/play-java-websocket-example/issues/51

Compare with https://github.com/playframework/play-scala-websocket-example/pull/74 (scala version of websocket example)

### CODE REVIEW

File: test/controllers/FunctionalTest.java
Comment (richdougherty) [inline test/controllers/FunctionalTest.java]: Delete this?

**Review by richdougherty — APPROVED** [review]
It looks good to me, but you might also want to get a review from someone who knows this app and streams a bit better. :)

File: test/controllers/FunctionalTest.java
Comment (wsargent) [reply test/controllers/FunctionalTest.java]: Done.  BTW, I lost the ability to merge PRs on the examples somewhere down the line (github status altered?)

Comment (richdougherty) [reply test/controllers/FunctionalTest.java]: I'll check up on your Github status for the examples. If you like you can email me if there's anything you think you should have access to that you don't and I'll sort it out. :)

**Comment by marcospereira** [comment]
Will, I'm getting the following error if I close the browser tab:

```
[error] a.a.RepointableActorRef - Error in stage [akka.stream.scaladsl.MergeHub$$anon$2@164b5b45]: Upstream producer failed with exception, removing from MergeHub now
akka.stream.scaladsl.MergeHub$ProducerFailed: Upstream producer failed with exception, removing from MergeHub now
	at akka.stream.scaladsl.MergeHub$$anon$2$$anon$4.onUpstreamFailure(Hub.scala:261)
	at akka.stream.impl.fusing.GraphInterpreter.processEvent(GraphInterpreter.scala:485)
	at akka.stream.impl.fusing.GraphInterpreter.execute(GraphInterpreter.scala:368)
	at akka.stream.impl.fusing.GraphInterpreterShell.runBatch(ActorGraphInterpreter.scala:571)
	at akka.stream.impl.fusing.GraphInterpreterShell$AsyncInput.execute(ActorGraphInterpreter.scala:457)
	at akka.stream.impl.fusing.GraphInterpreterShell.processEvent(ActorGraphInterpreter.scala:546)
	at akka.stream.impl.fusing.ActorGraphInterpreter.akka$stream$impl$fusing$ActorGraphInterpreter$$processEvent(ActorGraphInterpreter.scala:725)
	at akka.stream.impl.fusing.ActorGraphInterpreter$$anonfun$receive$1.applyOrElse(ActorGraphInterpreter.scala:740)
	at akka.actor.Actor.aroundReceive(Actor.scala:513)
	at akka.actor.Actor.aroundReceive$(Actor.scala:511)
Caused by: java.util.concurrent.TimeoutException: No demand signalled in the last 1 second.
	at akka.stream.impl.Timers$BackpressureTimeout$$anon$4.onTimer(Timers.scala:133)
	at akka.stream.stage.TimerGraphStageLogic.onInternalTimer(GraphStage.scala:1228)
	at akka.stream.stage.TimerGraphStageLogic.$anonfun$getTimerAsyncCallback$1(GraphStage.scala:1217)
	at akka.stream.stage.TimerGraphStageLogic.$anonfun$getTimerAsyncCallback$1$adapted(GraphStage.scala:1217)
	at akka.stream.impl.fusing.GraphInterpreter.runAsyncInput(GraphInterpreter.scala:443)
	at akka.stream.impl.fusing.GraphInterpreterShell$AsyncInput.execute(ActorGraphInterpreter.scala:453)
	at akka.stream.impl.fusing.GraphInterpreterShell.processEvent(ActorGraphInterpreter.scala:546)
	at akka.stream.impl.fusing.ActorGraphInterpreter.akka$stream$impl$fusing$ActorGraphInterpreter$$processEvent(ActorGraphInterpreter.scala:725)
	at akka.stream.impl.fusing.ActorGraphInterpreter$$anonfun$receive$1.applyOrElse(ActorGraphInterpreter.scala:740)
	at akka.actor.Actor.aroundReceive(Actor.scala:513)
```
We should "fail" more gracefully here, I think.

File: test/controllers/FunctionalTest.java
Comment (richdougherty) [reply test/controllers/FunctionalTest.java]: @wsargent, I've now given you (and the rest of the _Core_ team in Github) Write access to all example repos. Let me know how you get on next time you try to merge.

**Comment by wsargent** [comment]
From 

> 	at akka.stream.impl.Timers$BackpressureTimeout$$anon$4.onTimer(Timers.scala:133)

that is the backpressure timeout -- I removed it.

**Pull request merged**
